### PR TITLE
LibWeb: Fix bug where viewing resource schemed directory urls crashes

### PR DESCRIPTION
**Pull request by bplaat**
This pr removes the `@resource_directory_url@` template variable and uses the new `resource://` scheme to get to the right icons.

It also adds some code that when viewing a resource schemed directory url a generated directory page is responded instead of the browser crashing on a `VERIFY_NOT_REACH()` in `Resource.cpp`.

I also get the path to the `directory.html` and `error.html` page with resources to reduce code.

### CODE REVIEW

File: Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp [hunk 1]
```diff
@@ -89,7 +89,7 @@ ErrorOr<String> load_file_directory_page(LoadRequest const& request)
     StringBuilder builder;
     SourceGenerator generator { builder };
     generator.set("path", escape_html_entities(lexical_path.string()));
-    generator.set("parent_path", escape_html_entities(lexical_path.parent().string()));
+    generator.set("parent_url", TRY(String::formatted("file://{}", lexical_path.parent().string())));
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp:69]: Are we sure we don't need to `escape_html_entities` any longer?

File: Userland/Libraries/LibWeb/Loader/ResourceLoader.cpp [hunk 2]
```diff
@@ -250,6 +250,23 @@ void ResourceLoader::load(LoadRequest& request, SuccessCallback success_callback
             return;
         }
 
+        // // When resource uri is a directory use file directory loader to generate response
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/Loader/ResourceLoader.cpp:253]: You have 2 `//` here

Comment (trflynn89) [reply Userland/Libraries/LibWeb/Loader/ResourceLoader.cpp:253]: Please also capitalize "URI" here, and also capitalize "URL" in the commit message

File: Ladybird/Android/src/main/cpp/WebContentService.cpp [hunk 3]
```diff
@@ -72,9 +72,6 @@ ErrorOr<int> service_main(int ipc_socket, int fd_passing_socket)
     Web::HTML::Window::set_internals_object_exposed(is_layout_test_mode);
     Web::Platform::FontPlugin::install(*new Ladybird::FontPlugin(is_layout_test_mode));
 
-    Web::set_error_page_url(TRY(String::formatted("file://{}/res/ladybird/error.html", s_serenity_resource_root)));
```
Comment (trflynn89) [inline Ladybird/Android/src/main/cpp/WebContentService.cpp:76]: Commit message - please capitalize "HTML"

**Review by trflynn89 — CHANGES_REQUESTED** [review]
Beat me to it, great!

File: Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp (hunk 1 above)
Comment (bplaat) [reply Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp:69]: No because this variable is only used in href attr

**Comment by bplaat** [comment]
Thanks for the review Flynn, I've fixed your remarks

File: Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp:69]: Why's that matter? That was also the case before this patch. If the string contains a quote or an opening/closing brace, we would generate invalid HTML.

Comment (bplaat) [reply Userland/Libraries/LibWeb/Loader/GeneratedPagesLoader.cpp]: Sure, my thoughts where then the URL is invalid but yea, adding it back won't hurt

**Pull request merged**